c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2742,18 +2770,27 @@ class DeviceCachingAllocator {
     C10_CUDA_CHECK(c10::cuda::MaybeSetDevice(prev_device));
   }
 
-  void insert_events_deferred_until_no_capture() {
+  void insert_events_deferred_until_no_capture(
```

Comment:
This function originally inserted cuda events for things that recorded streams while cuda capturing. But now those same events won't get inserted anyway because they will be marked in block_to_cudagraph_stream_uses. Can't this be done with simpler logic that just records set_of_blocks_with_any_record_stream_during_graph_capture, and then puts blocks that are in that set and freed during graph capture into a list of Blocks to free during insert_events_deferred_until_no_capture? There doesn't seem to be a need to record the original streams or put the events in afterward.